docs/audit_report/src/06_bibliography.rst

```diff
@@ -0,0 +1,83 @@
+.. only:: not latex
+
+   Bibliography
+   ============
+
+.. [PRM] René Fischer, Juraj Somorvsky, Daniel Neus, Phillippe Lieser, René Meusel:
+   "Pflege und Weiterentwicklung der Kryptobibliothek Botan (Weiterentwicklung Botan):
+   Prüfmethodik für die Freigabe neuer Botan-Versionen",
+   Release TBD
+
+.. [CRYPD] René Fischer, Juraj Somorovsky, Tobias Niemann, Fabian Weißberg,
+   Sergii Cherkavskyi, Philippe Lieser, René Meusel, Amos Treiber, Fabian Albert:
+   "Pflege und Weiterentwicklung der Kryptobibliothek Botan (Weiterentwicklung Botan):
+   Cryptographic Documentation",
+   Release TBD
+
+.. [REFG] Jack Lloyd, Daniel Neus, René Fischer, Juraj Somorovsky, Tobias Niemann:
+   "Botan Reference Guide",
+   Release 3.1.1
+
+.. [ARCH] "Pflege und Weiterentwicklung der Kryptobibliothek Botan (Weiterentwicklung Botan):
+   Botan Prüfung 3.1.1 - Library Architecture Overview",
+   Release TBD
+
+.. [TESTSP] "Pflege und Weiterentwicklung der Kryptobibliothek Botan (Weiterentwicklung Botan):
+   Test Specification",
+   Release TBD
+
+.. [TESTRP] "Pflege und Weiterentwicklung der Kryptobibliothek Botan (Weiterentwicklung Botan):
+   Botan Test Report",
+   Release TBD
+
+.. [BOTAN_GIT] https://github.com/randombit/botan/tree/3.1.1
+
+.. [DATA] https://www.usenix.org/conference/usenixsecurity18/presentation/weiser
+
+.. [DATA_GIT] https://github.com/Fraunhofer-AISEC/DATA
+
+.. [INTEL_PIN] https://www.intel.com/content/www/us/en/developer/articles/tool/pin-a-dynamic-binary-instrumentation-tool.html
+
+.. [TVLA] https://csrc.nist.gov/CSRC/media/Events/Non-Invasive-Attack-Testing-Workshop/documents/08_Goodwill.pdf
+
+.. [RDC] https://proceedings.neurips.cc/paper/2013/file/aab3238922bcc25a6f606eb525ffdc56-Paper.pdf
+
+.. [CT_GRIND] https://github.com/agl/ctgrind
+
+.. [BOTAN_CT_UTILS] https://github.com/randombit/botan/blob/master/src/lib/utils/ct_utils.h
+
+.. [VALGRIND_MEMCHECK] https://valgrind.org/docs/manual/mc-manual.html
+
+.. [BOTAN_TIMING_TEST] https://github.com/randombit/botan/blob/master/src/cli/timing_tests.cpp
+
+.. [MONA_TIMING] https://github.com/seecurity/mona-timing-report
+
+.. [BOTAN_KYBER_KEM_CRYPTOR] https://github.com/randombit/botan/blob/3.0.0/src/lib/pubkey/kyber/kyber_common/kyber.cpp#L1228
+
```

Comment:
In the meantime we added a Sphinx extension that simplifies links to the Botan sources. This allows referring to source files by simply using a Sphinx role inline with the text:

```rst
:srcref:`src/lib/pubkey/kyber/kyber_common/kyber.cpp`
```

The extension also allows to check for dead links, which is rather handy.

One caveat: currently it does not support line numbers. @lieser Do you think we should use the `:srcref:` extension for those? Maybe even add support for line numbers? I'd be concerned that they are harder to maintain when moving to another Botan version. However, the audit report document is usually tacked to a specific Botan version, so that might actually be okay for this use case.

Note: The extension isn't currently enabled for this document.

Reply:
I tend to say just adding support for line numbers to `:srcref:` is probably not a good idea.
Unless we extend it at the same time to also allow to specify the ref to use instead of using the global one.

If we use line numbers with the global ref, it will be easy to forget to update the line number, and have a wrong one.

In general we should avoid line numbers there possible, and only add them if they are required for clarification.

Reply:
Yeah. I share your views and concerns.